llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -918,6 +918,44 @@ bool llvm::hasIterationCountInvariantInParent(Loop *InnerLoop,
   return true;
 }
 
+constexpr Intrinsic::ID llvm::getReductionIntrinsicID(RecurKind RK) {
```

Comment:
This function seems to only be used to create a VP reduction. Why can't we rename this to getVPReductionIntrinsicID and just return the VP intrinsic ID directly? Why do we need to convert from non-VP to VP?

Reply:
Is this because we're trying to copy the `createVectorInstruction` design where we pass the non-VP opcode?

Reply:
I guess we will need this functionality for EVL vectorization of other intrinsics so maybe this is the right design.